libarchive/archive_read_support_format_iso9660.c

```diff
@@ -951,6 +979,51 @@ isPVD(struct iso9660 *iso9660, const unsigned char *h)
 	return (48);
 }
 
+static int
+isRootDirectoryRecord(const unsigned char *p) {
```

Comment:
Here's the fun part.  Remember, we want to reject anything that is definitely not a root directory record.